### PR TITLE
3.2, 3.3, 3.4*

### PR DESCRIPTION
**Pull request by ashleve**

### CODE REVIEW

**Comment by ashleve** [comment]
> Direct coupling with DrawerFeature class is unnecessary.

> Hardcoded line type. What about 3.3.2 test?
> Unfortunately, this is not a generic solution. (e.g. new line type requires new class, unnecessary direct coupling)

I fixed the issues.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    private DrawPanelController controller;
+    private ILine line;
+    private int startX = 0, startY = 0;
+
+    public LineDrawerAdapter(DrawPanelController controller, LineType lineType) {
+        this.controller = controller;
+
+        switch(lineType){
+            case BASIC:
+                this.line = LineFactory.getBasicLine();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDrawerAdapter.java:20]: Coupling with LineFactory and LineType not necessary. KISS principle.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/MyAdapter.java [hunk 2]
```diff
@@ -26,8 +27,9 @@ public void operateTo(int x, int y) {
 		ILine line = LineFactory.getBasicLine();
 		line.setStartCoordinates(this.startX, this.startY);
 		line.setEndCoordinates(x, y);
+		setPosition(x, y);
 
-		drawLine(line);
+		getDrawerController().drawLine(line);
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/MyAdapter.java:32]: Constructor injection should be used (using a parameter). This is unnecessary direct coupling with DrawerFeature class.

File: src/main/java/edu/kis/powp/jobs2d/drivers/adapter/MyAdapter.java [hunk 3]
```diff
@@ -4,11 +4,12 @@
 import edu.kis.legacy.drawer.shape.ILine;
 import edu.kis.legacy.drawer.shape.LineFactory;
 import edu.kis.powp.jobs2d.Job2dDriver;
+import edu.kis.powp.jobs2d.features.DrawerFeature;
 
 /**
  * driver adapter to drawer with several bugs.
  */
-public class MyAdapter extends DrawPanelController implements Job2dDriver {
+public class MyAdapter extends DrawerFeature implements Job2dDriver {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/adapter/MyAdapter.java:12]: Name not fixed

File: src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java [hunk 4]
```diff
@@ -16,6 +17,14 @@ public SelectTestFigureOptionListener(DriverManager driverManager) {
 
 	@Override
 	public void actionPerformed(ActionEvent e) {
-		FiguresJoe.figureScript1(driverManager.getCurrentDriver());
+		switch (e.getActionCommand()){
+			case "Figure Joe 1":
```
Comment (kowallus) [inline src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java:21]: Menu action should not depend on its caption. Enum or separate listeners would be better. At least you should consider constants.

**Review by kowallus — COMMENTED** [review]
4 pts

Specific remarks are inlined in the code.
UML diagram is missing

**Pull request closed without merging**